File: src/com/suushiemaniac/cubing/alglib/move/modifier/MegaminxWideModifier.java
package com.suushiemaniac.cubing.alglib.move.modifier;

public enum MegaminxWideModifier implements MegaminxModifier {
    PLUS("++"), MINUS("--");

    public static MegaminxWideModifier fromNotation(String notation) {
        for (MegaminxWideModifier modifier : MegaminxWideModifier.values())
            if (modifier.getNotation().equals(notation)) return modifier;
        return null;
    }

    private String notation;

    MegaminxWideModifier(String notation) {
        this.notation = notation;
    }

    public String getNotation() {
        return this.notation;
    }

    @Override
    public String toFormatString() {
        return this.getNotation();
    }

    @Override
    public MegaminxWideModifier inverse() {
        return MegaminxWideModifier.values()[(this.ordinal() + 1) % MegaminxWideModifier.values().length];
    }

    @Override
    public MegaminxWideModifier merge(Modifier other) {
        return null;
    }

    @Override
    public String getType() {
        return "WIDE";
    }
}

File: src/main/kotlin/com/suushiemaniac/cubing/alglib/move/modifier/MegaminxWideModifier.kt
package com.suushiemaniac.cubing.alglib.move.modifier

enum class MegaminxWideModifier(override val notation: String) : MegaminxModifier {
    PLUS("++"), MINUS("--");

    override val type = "WIDE"

    override fun inverse() = MegaminxWideModifier.values()[(this.ordinal + 1) % MegaminxWideModifier.values().size]
    override fun merge(other: Modifier) = other as MegaminxWideModifier
}